torch_geometric/nn/aggr/attention.py

```diff
@@ -45,13 +45,21 @@ def reset_parameters(self):
         reset(self.gate_nn)
         reset(self.nn)
 
-    def forward(self, x: Tensor, index: Optional[Tensor] = None,
-                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
-                dim: int = -2) -> Tensor:
+    def forward(
+        self,
+        x: Tensor,
+        index: Optional[Tensor] = None,
+        ptr: Optional[Tensor] = None,
+        dim_size: Optional[int] = None,
+        dim: int = -2,
+        batch: Optional[Tensor] = None,
+        batch_size: Optional[int] = None,
+    ) -> Tensor:
 
         self.assert_two_dimensional_input(x, dim)
-        gate = self.gate_nn(x)
-        x = self.nn(x) if self.nn is not None else x
+        gate = self.gate_nn(x, batch=batch, batch_size=batch_size)
```

Comment:
@rusty1s I'm not sure if `self.gate_nn` and `self.nn` are typed as `Module` in the documentation because they're supposed to be anything other than `MLP` or because you needed to avoid the circular import error. The papers discussing attentional aggregation describe H gate (gate_nn) and H theta (nn) as MLPs.  To avoid a type error, we'd need to clarify in the documentation that the expectation is for them to be `MLP`s despite the more generic `Module` label.

Reply:
Yes, it doesn't necessarily need to be a `torch_geometric.nn.MLP`. I changed this accordingly.